### PR TITLE
refactor(runtime-fuzzer): Reduce fuzzer input debug printed data when crash happens

### PR DESCRIPTION
**Pull request by SabaunT**

### CODE REVIEW

File: utils/runtime-fuzzer/src/lib.rs [hunk 1]
```diff
@@ -33,8 +33,40 @@ use gear_runtime::{AccountId, Gear, Runtime, RuntimeOrigin};
...
+}
+
+impl<'a> RuntimeFuzzerInput<'a> {
+    pub fn into_inner(self) -> &'a [u8] {
+        self.0
+    }
+}
+
 /// Runs all the fuzz testing internal machinery.
 pub fn run(data: &[u8]) -> Result<()> {
```
Comment (mertwole) [inline utils/runtime-fuzzer/src/lib.rs:71]:
```suggestion
pub fn run(data: RuntimeFuzzerInput) -> Result<()> {
```

**Review by mertwole — APPROVED** [review]

**Review by Magorsky — APPROVED** [review]

File: utils/runtime-fuzzer/src/lib.rs [hunk 2]
```diff
@@ -33,8 +33,40 @@ use gear_runtime::{AccountId, Gear, Runtime, RuntimeOrigin};
...
+/// therefore to avoid printing huge amount of data we do a mock implementation of [`Debug`].
+impl Debug for RuntimeFuzzerInput<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("RuntimeFuzzerInput")
+            .field(&"Mock `Debug` impl")
+            .finish()
+    }
+}
+
+impl<'a> RuntimeFuzzerInput<'a> {
```
Comment (breathx) [inline utils/runtime-fuzzer/src/lib.rs:64]: maybe just as ref?

File: utils/runtime-fuzzer/src/lib.rs [hunk 3]
```diff
@@ -33,8 +33,40 @@ use gear_runtime::{AccountId, Gear, Runtime, RuntimeOrigin};
...
 
+/// This is a wrapper over random bytes provided from fuzzer.
+///
+/// It's main purpose is to be a mock implementor of `Debug`.
+/// For more info see `Debug` impl.
+pub struct RuntimeFuzzerInput<'a>(&'a [u8]);
+
+impl<'a> Arbitrary<'a> for RuntimeFuzzerInput<'a> {
+    fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self> {
+        let ret = Self(u.peek_bytes(u.len()).expect("took bytes within buffer len"));
```
Comment (breathx) [inline utils/runtime-fuzzer/src/lib.rs:47]: not intuitive expect panic message

**Review by breathx — APPROVED** [review]

File: utils/runtime-fuzzer/src/lib.rs (hunk 2 above)
Comment (SabaunT) [reply utils/runtime-fuzzer/src/lib.rs:64]: after this there will be no need in `into_inner`

File: utils/runtime-fuzzer/src/lib.rs (hunk 1 above)
Comment (SabaunT) [reply utils/runtime-fuzzer/src/lib.rs]: done

File: utils/runtime-fuzzer/src/lib.rs (hunk 3 above)
Comment (SabaunT) [reply utils/runtime-fuzzer/src/lib.rs]: changed to return an error

**Pull request merged**